.github/workflows/twister_dfu_nrf52840dk.yml

```diff
@@ -0,0 +1,138 @@
+name: Twister DFU nrf52840dk
+
+on:
+  push:
+    branches: [ main ]
+  schedule:
+    # Run workflow at the start of every day (12 AM UTC)
+    - cron: "0 0 * * *"
+
+  workflow_dispatch:
+
+jobs:
+  # Assumptions made about the self-hosted runner:
+  #
+  #  1. Has installed the GitHub Actions self-hosted runner service
+  #  2. Has nrfjprog already installed at /opt/nrf-command-line-tools
+  #  3. Has SEGGER tools already installed at /home/golioth/JLink_Linux_V786_arm64
+  #  4. GitHub secret YOUR_RUNNER_NAME_NRF52840DK_PORT
+  #  5. GitHub secret YOUR_RUNNER_NAME_NRF52840DK
+  #  6. GitHub secret GOLIOTHCTL_YAML
+  #
+  # It is the responsibility of the self-hosted runner admin to ensure
+  # these pre-conditions are met.
+  #
+  # For item 1, GitHub provides instructions when you add a new self-hosted runner
+  # in Settings -> Actions -> Runners.
+  #
+  # For item 2, you'll need to install jlink and nrf-command-line-tools:
+  #   https://www.segger.com/downloads/jlink/
+  #   https://www.nordicsemi.com/Products/Development-tools/nrf-command-line-tools/download
+  #
+  # For item 3, this environment variable can be added to $HOME/runner_env.sh:
+  #   export CI_NRF52840DK_PORT=/dev/ttyACM0
+  #
+  # For item 4, the file needs to have contents like (use base64 encoding):
+  #
+  # - connected: true
+  #   id: "001050257254"
+  #   platform: nrf52840dk_nrf52840
+  #   product: J-Link
+  #   runner: nrfjprog
+  #   serial: /dev/serial/by-id/usb-SEGGER_J-Link_001050257254-if00
```

Comment:
This is runner specific, so IMO it will be better to store it on the machine connected to DUT, e.g. at `/etc/hardware-map.yaml` and then mapped into the container with another volume:
```
      volumes:
        - /etc/hardware-map.yaml:/etc/hardware-map.yaml
```

Reply:
The OrangePis used for self-hosted runners are not easy to access. For that reason it is better to make all configuration files exist as GitHub Secrets. I have an improvement in mind that I will outline in a design document during next sprint. It will outline a process where new hardware can be added to a runner without any local configuration. I recommend going with what we have here for this PR and then refining as we review that design doc.

Reply:
Well, hardware configuration depends on what is connected to the runner. This is something that needs to be checked manually, e.g. by serial connection with Linux or with SSH. And the only way you know what is connected is by checking it physically (in order to know which DevKit/board is connected). Another thing is `/dev/serial/*` node to be used. Last thing is actually the list of connected boards (there might be many of them connected to a single runner).

I really don't see why we need to store "hardware map" (information about what is connected to a single runner) in a central database (which is GitHub Actions Secrets).

Reply:
It's a pain to get into the shell of these devices. They are running on their own network, so you need to connect to that network and then ssh into them. At that point the person who is running it needs to know enough about how they're set up to find the device serial endpoint and then know enough about formatting the hardware map files to create one. Then run a test using that file. It's a lot of overhead for people who don't work with these systems often. I've been working with these self-hosted runners for quite a while now and it is still a pain for me to do this.

My idea is to have a workflow that will get the output of `ls -la /dev/serial/by-id` which can then be used to create the hardware map. This way, adding new hardware means just plugging it in and running the workflow.

I think we should continue this discussion once I have submitted the design document for this.